[JBIDE-23629] fixed AddConnectionTest to lookup conns by host

adietish opened the pull request:
Connections in the Docker explorer must be unique in name and unique in
host. AddConnectionTest thus can fail if there already is a connection
using the same host with a different name.
This should be fixed by looking up connections by their host.

adietish commented:
@jkopriva please review

adietish commented:
this depends on #1692 which should be merge before

adietish commented:
@jkopriva I rebased to latest origin/master (with your changes). Can you please review this PR?

adietish commented:
@jkopriva #1692 was merged and I included the changes that we discussed, can you please review this PR?

jkopriva commented on tests/org.jboss.tools.docker.ui.bot.test/src/org/jboss/tools/docker/ui/bot/test/AbstractDockerBotTest.java:
```diff
@@ -64,113 +72,126 @@ protected static void cleanupShells() {
+			this.connection = getConnectionByHost(dockerServerURI);
+		} else if (!StringUtils.isEmpty(unixSocket)) {
 			dockerView.createDockerConnection(AuthenticationMethod.UNIX_SOCKET, unixSocket, null, unixSocket);
+			this.connection = getConnectionByHost(dockerServerURI);
```
there should be unixSocket, instead of dockerServerURI?

jkopriva commented:
@adietish can you please also fix class org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/CDKDevstudioBaseTest.java ?
There is a compilation error (https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/jbt-integration-tests-prcheck/jdk=jdk1.8,label=rhel7/103/console), just fix the compilation error, that it can be build and the PR can be verified by Jenkins job. @odockal will make adjustments (if needed) in another PR.

adietish commented:
@jkopriva compilation error fixed here: https://github.com/jbosstools/jbosstools-integration-tests/pull/1696/files#diff-7c0c1683eafb995e57593db505ee07a5L228 and above bug corrected (dockerServerURI replaced with unixSocket)

jkopriva commented:
testPR

adietish commented:
@jkopriva I removed the AbstractDockerBotTest#openDockerPerspective and replaced it with @OpenPerspective(DockerPerspective.class) and also remove AbstractDockerBotTest#executeCommand as discussed. Please re-review

jkopriva commented:
testPR

adietish commented:
testPR

adietish commented:
testPR
